classes/ObjectModel.php

```diff
@@ -1149,27 +1149,39 @@ public function validateField($field, $value, $id_lang = null, $skip = [], $huma
 
         // Check field validator
         if (!in_array('validate', $skip) && !empty($data['validate'])) {
-            if (!method_exists('Validate', $data['validate'])) {
-                throw new PrestaShopException($this->trans('Validation function not found: %s.', [$data['validate']], 'Admin.Notifications.Error'));
+            $callable = $data['validate'];
+            if (!is_callable($callable)) {
+                $callable = 'Validate::'. $data['validate'];
+            }
+
+            if (!is_callable($callable)) {
+                throw new PrestaShopException(
+                    $this->trans(
+                        'Validation function not found: %s.',
+                        array($data['validate']),
+                        'Admin.Notifications.Error'
+                    )
+                );
             }
 
             if (!empty($value)) {
-                $res = true;
-                if (Tools::strtolower($data['validate']) == 'iscleanhtml') {
+                $is_valid = true;
+                if (Tools::strtolower($data['validate']) === 'iscleanhtml') {
                     if (!call_user_func(['Validate', $data['validate']], $value, $ps_allow_html_iframe)) {
-                        $res = false;
-                    }
-                } else {
-                    if (!call_user_func(['Validate', $data['validate']], $value)) {
-                        $res = false;
+                        $is_valid = false;
                     }
                 }
-                if (!$res) {
+
+                if (!call_user_func($callable, $value)) {
```

Comment:
Same here: I'm afraid php is not really clear and strict on what can be called and what cannot be called.
You replaced `call_user_func(['Validate', $data['validate']], $value)` by `call_user_func(Validate::'. $data['validate'], $value`
It looks equivalent but honestly I'm not a php expert so I cannot say it covers 100% possible php usages and these 2 ways of writing are equivalent. I would feel safer if we keep previous notation to ensure we dont modify the behavior.

Reply:
=> https://www.php.net/manual/fr/language.types.callable.php

In any case, this will works as the previous check (remember : `is_callable`) is true.

Fear something is never a good reason for not improving something: if I redo that change, I'll need at least 3 additional if/else statments in this function: this is not something I want to write  (or test).

I know it because ... it was the first version of my feature :)

Reply:
If I read correctly https://www.php.net/manual/en/language.types.callable.php (EN version) then `call_user_func(Validate::'. $data['validate'], $value)` handles less usecases than `call_user_func(['Validate', $data['validate']], $value)`.

See the 1st snippet of code "Example 1 Callback function examples"
```
call_user_func(array('B', 'parent::who')); // A
```
If `$data['validate']` is a string `parent::a_function_name` then your concatenation will convert it to `Validate::parent::a_function_name` which I dont think works 🤔 

Again, I'm sorry for being paranoid this PR modifies something core of PrestaShop 😄extra caution is mandatory here. And since we did not defined [a clear contract](https://build.prestashop.com/news/prestashop-in-2019-and-beyond-part-2-pain-points/#no-clearly-defined-api) for PrestaShop API we have to make sure we preserve all existing behaviors following SemVer.

Reply:
>  we have to make sure we preserve all existing behaviors following SemVer.

This is pretty simple! the only thing that works now is when you pass the name of the function as value it looks for available (public) methods in `Validate` class.

* Current behavior when using `parent::a_function_name` => throw a PrestaShop Exception
* New behavior when using `parent::a_function_name` => throw a PrestaShop Exception

Why ?

_Because of the method_exists check_: https://3v4l.org/dOskN